### ISSUE TITLE
Java 8 compatibility

### ISSUE THREAD

**Issue by youribonnaffe** [FroMage/jax-doclets#61]
It seems that with Java 8 every is broken.

I started to do some changes but it feels like this is going to be a lot of rework, the doclet API has been through massive changes..

Unfortunately I cannot find a lot of details about this API online. I'd like to contribute but I wondering what the best solution would be.

**Comment by FroMage** [comment]
Aw, damn. That would also mean we need to break Java < 8 compatibility if we support it, no?

**Comment by youribonnaffe** [comment]
It seems so. 
The options I see for now are: 
- Java 8 only version
- embed doclet API from Java < 8 and rename packages (dirty)

What bothers me is that I quickly looked for information about API changes for doclets and did not find anything. Isn't it used a bit, by other people writing custom doclets?...

**Comment by FroMage** [comment]
I don't think anybody writes doclets anymore ;)

**Comment by wikier** [comment]
Same issue in Apache Marmotta ([MARMOTTA-513](https://issues.apache.org/jira/browse/MARMOTTA-513)), where the doclet is used to generate documentation form the JAX-RS web services.

Any plan to support `Java>=8` ?

**Comment by FroMage** [comment]
The truth of the matter is that I don't have time to maintain this project anymore. If you want to make it work for both Java 7 and Java 8 I can grant you commit access.

**Comment by wikier** [comment]
For us (Marmotta) the urgent think would be to get 0.10.1 released and pushed to maven central as soon as possible,

I have no time either to maintain it, but I do have some time to keep it running and manage some fixes.

**Comment by FroMage** [comment]
Well, I'm not using this project for work anymore, which is also why I don't have time to maintain it. If you have time to put at least your fixes, and perhaps other contributors's fixes that work for you, I can let you push to maven central too (if I figure out how that works again). Let me know if that would work for you and I'll arrange it.

**Comment by wikier** [comment]
If you give me the credentials, I can help to get 0.10.1 out (issue #63).

**Comment by FroMage** [comment]
You should have push access now. For Maven Central it's trickier: I just checked and you need to create a JIRA to ask for permission to publish jax-doclets like me. I can't grant you that permission in the UI myself.

**Comment by wikier** [comment]
@FroMage, are you publishing via Sonatype OSS, right? Then you should open an issue, referencing [OSSRH-855](https://issues.sonatype.org/browse/OSSRH-855), asking to grant me (`wikier`) over the `com.lunatech.jax-doclets` groupId.

Please, comment issue #63 whatever should be taken into account for publishing 0.10.1 (what contains, etc)

**Comment by FroMage** [comment]
OK done: https://issues.sonatype.org/browse/OSSRH-10624

**Comment by FroMage** [comment]
As for what went in, you can see the list here: https://github.com/FroMage/jax-doclets/issues?direction=desc&milestone=2&page=1&sort=created&state=closed and what should go in, I don't really care. Release when you need it released ;) Releasing often is never a bad thing.

**Comment by FroMage** [comment]
Permission granted apparently.

**Comment by wikier** [comment]
continue this thread at issue #63

**Comment by dpalic** [comment]
I am wondering if this project is still alive? This issue is very old and still not closed.
Also JDK7 has this month it last month about any public updates. So the pressure is here to move to JDK8. Can someone tell more about the project status and about any further interest on the project?

**Comment by wikier** [comment]
not really... I jumped in to solve a concrete an quick issue, but @FroMage is not maintaining it anymore, I guess.

**Comment by FroMage** [comment]
No, this is pretty much looking for a brave maintainer :)

**Comment by dpalic** [comment]
Is there any idea of how many users are using this doclet project? I am willing to support, but only if a community is in the background to help us too...

**Comment by FroMage** [comment]
I've no idea TBH.

**Comment by wikier** [comment]
We use it in Apache Marmotta, that's why I needed released 0.10.1 (see issue #63). So I may help.

**Comment by dpalic** [comment]
FroMage, are you willing to support us with your knowledge? If so, we are already 3 persons to maintain ;-)

**Comment by FroMage** [comment]
What little knowledge I have I will share gladly ;)
